### DISCUSSION TITLE
Why are the -text, -bg-subtle and -border-subtle variables not using the key variable as their root?

### DISCUSSION THREAD

**Discussion by Nysosis** [twbs/bootstrap#37911] [General]
I'm just looking through the https://github.com/twbs/bootstrap/blob/main/scss/_variables.scss and wondering why the variables at https://github.com/twbs/bootstrap/blob/main/scss/_variables.scss#L325 aren't based off the actual, and primarily I'm concerned on `$primary` variable, instead they use the appropriate shade of what the default colour is.

If I change the $primary, I would have thought it would feed through into all derived colours, instead of also needing to override those (and potentially others down the line)

**Comment by yugabe** [comment]
I'm having problems with configuring colors correctly as well. And I think it's not a "me" problem.

Take this for example:
```scss
$primary-text-dark:                 $blue-300 !default;
```

It should be instead, clearly:
```scss
$primary-text-dark:                  tint-color($primary, 40%) !default;
```

This is repeated through the whole of the project.

https://github.com/twbs/bootstrap/blob/1299b163c7526b0e6892856d87e8710bc195d7f5/scss/_variables-dark.scss#L11

```scss
$primary-text-emphasis-dark:        $blue-300 !default;
```

Good luck to anyone trying to configure colors in 5.3.0. They just copy-pasted $blue everywhere. Thanks.

**Reply by mdo** [reply]
Yup, you're right, not ideal. Originally I thought about these being disconnected on purpose, because how would you know if you needed to `tint()` or `shade()` a particular color, but their purposes are generally pretty obvious now. Text emphasis is darker, subtle background is lighter. Going to stub out a branch and PR to make this happen. Appreciate the honest and direct feedback y'all!

**Reply by yugabe** [reply]
Hey @mdo, I'm sorry to ping you here too, but to avoid potentially doing double work, please take a look at the issue I opened regarding this here: https://github.com/twbs/bootstrap/issues/37949

It's essentially a mini-PR containing code that might be a good jumping off point for this.
